internal/pkg/service/common/dependencies/distlock.go

```diff
@@ -0,0 +1,42 @@
+package dependencies
+
+import (
+	"context"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/distlock"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+// distributedLockScope implements DistributedLockScope interface.
+type distributedLockScope struct {
+	provider *distlock.Provider
+}
+
+type distributedLockScopeDeps interface {
+	BaseScope
+	EtcdClientScope
+}
+
+func NewDistributedLockScope(ctx context.Context, cfg distlock.Config, d distributedLockScopeDeps) (DistributedLockScope, error) {
+	return newDistributedLockScope(ctx, cfg, d)
+}
+
+func newDistributedLockScope(ctx context.Context, cfg distlock.Config, d distributedLockScopeDeps) (v *distributedLockScope, err error) {
+	ctx, span := d.Telemetry().Tracer().Start(ctx, "keboola.go.common.dependencies.NewDistributedLockScope")
+	defer span.End(&err)
+
+	provider, err := distlock.NewProvider(cfg, d)
```

Comment:
I would check for an error.

Reply:
:+1: thaks, fixed https://github.com/keboola/keboola-as-code/pull/1594/commits/2fe60d96befb393130c630b5a6ead8c5e354b2c3